albatross/core/indexing.h

```diff
@@ -47,6 +54,21 @@ inline Eigen::VectorXd subset(const std::vector<SizeType> &indices,
   return out;
 }
 
+/*
+ * Extracts a subset of columns from an Eigen::Matrix
+ */
+template <typename SizeType>
```

Comment:
Does this have to be templated? can it not just be hardcoded to use `size_t` or `unsigned` or whatever?

Reply:
Yeah this might be overuse of templating, in general the value type of `col_indices` can be anything that can be converted to an `Eigen::Index`, but now I think the only thing that is really used is `s32`.  Whatever the decision is the same argument would apply to all the rest of the `subset` functions in this file as well.